server/grpc/api/instance.go

```diff
@@ -23,12 +23,17 @@ func NewInstanceServer(sdk *sdk.SDK) *InstanceServer {
 
 // List instances.
 func (s *InstanceServer) List(ctx context.Context, request *protobuf_api.ListInstancesRequest) (*protobuf_api.ListInstancesResponse, error) {
-	hash, err := hash.Decode(request.ServiceHash)
-	if err != nil {
-		return nil, err
+	var (
```

Comment:
```go
h, err := hash.Decode(request.ServiceHash)
if request.ServiceHash != "" && err != nil {
  return err
}
```

It's a bit cleaner but this one is also ok :)

Reply:
This was done in https://github.com/mesg-foundation/engine/pull/1102, not in this PR.